frame/vesting/src/benchmarking.rs

```diff
@@ -24,73 +24,96 @@ use super::*;
 use frame_system::{RawOrigin, Pallet as System};
 use frame_benchmarking::{benchmarks, account, whitelisted_caller, impl_benchmark_test_suite};
 use sp_runtime::traits::Bounded;
+use frame_support::assert_ok;
 
 use crate::Pallet as Vesting;
 
 const SEED: u32 = 0;
 
-type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
 
 fn add_locks<T: Config>(who: &T::AccountId, n: u8) {
-	for id in 0..n {
+	for id in 0 .. n {
 		let lock_id = [id; 8];
-		let locked = 100u32;
+		let locked = 256u32;
 		let reasons = WithdrawReasons::TRANSFER | WithdrawReasons::RESERVE;
 		T::Currency::set_lock(lock_id, who, locked.into(), reasons);
 	}
 }
 
-fn add_vesting_schedule<T: Config>(who: &T::AccountId) -> Result<(), &'static str> {
-	let locked = 100u32;
-	let per_block = 10u32;
+fn add_vesting_schedules<T: Config>(
+	target: <T::Lookup as StaticLookup>::Source,
+	n: u32,
+) -> Result<BalanceOf<T>, &'static str> {
+	let min_transfer = T::MinVestedTransfer::get();
+	let per_block_duration_20 = min_transfer.checked_div(&20u32.into()).unwrap();
+
+	let locked = min_transfer;
+	let per_block = per_block_duration_20;
 	let starting_block = 1u32;
 
-	System::<T>::set_block_number(0u32.into());
+	let source: T::AccountId = account("source", 0, SEED);
+	let source_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(source.clone());
+	T::Currency::make_free_balance_be(&source, BalanceOf::<T>::max_value());
+
+	System::<T>::set_block_number(T::BlockNumber::zero());
 
-	// Add schedule to avoid `NotVesting` error.
-	Vesting::<T>::add_vesting_schedule(
-		&who,
-		locked.into(),
-		per_block.into(),
-		starting_block.into(),
-	)?;
-	Ok(())
+	let mut total_locked: BalanceOf<T> = Zero::zero();
+	for _ in 0 .. n {
+		total_locked += locked;
+
+		let schedule = VestingInfo::new::<T>(locked, per_block, starting_block.into());
+		assert_ok!(Vesting::<T>::do_vested_transfer(
+			source_lookup.clone(),
+			target.clone(),
+			schedule
+		));
+	}
+
+	Ok(total_locked.into())
 }
 
 benchmarks! {
 	vest_locked {
-		let l in 0 .. MaxLocksOf::<T>::get();
+		let l in 0 .. MaxLocksOf::<T>::get() - 1;
+		let s in 1 .. T::MaxVestingSchedules::get();
 
-		let caller = whitelisted_caller();
-		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+		let caller: T::AccountId = whitelisted_caller();
+		let caller_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(caller.clone());
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value() / 2u32.into());
```

Comment:
Why are we dividing by 2 here?

To be a little more precise, it would make sense to call `make_free_balance_be` inside the `add_vesting_schedules` loop, and top up the balance before each transfer.

You can imagine some pretty crazy settings that could cause the current code to fail, although very unlikely.

Reply:
a9fca6d4a5e97524dd9c2b319e5c98d533fa0d96

This is just the account that vest is being called on, but its not the account used in `add_vesting_schedules`.
It needs just enough balance to call vest, so I changed that to `minimum_balance`.

In `add_vesting_schedules` I also changed it so now the sending account tops off at the end of each loop, which should be more resillient against various configs.